src/Discord.Net/Entities/Emotes/IGuildEmote.cs

```diff
@@ -1,7 +1,44 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
 namespace Discord
 {
-    public interface IGuildEmote : IEmote
+    public interface IGuildEmote : IEmote, ISnowflakeEntity, IDeletable
     {
-        // TODO
+        /// <summary>
+        ///     Gets whether this emoji is managed by an integration.
+        /// </summary>
+        /// <returns>
+        ///     A boolean that determines whether or not this emote is managed by a Twitch integration.
+        /// </returns>
+        bool IsManaged { get; }
+        /// <summary>
+        ///     Gets whether this emoji must be wrapped in colons.
+        /// </summary>
+        /// <returns>
+        ///     A boolean that determines whether or not this emote requires the use of colons in chat to be used.
+        /// </returns>
+        bool RequireColons { get; }
+        /// <summary>
+        ///     Gets the roles that are allowed to use this emoji.
+        /// </summary>
+        /// <returns>
+        ///     A read-only list containing snowflake identifiers for roles that are allowed to use this emoji.
+        /// </returns>
+        IReadOnlyList<IRole> Roles { get; }
+        /// <summary>
+        ///     Gets the user ID associated with the creation of this emoji.
+        /// </summary>
+        /// <returns>
+        ///     An <see cref="ulong"/> snowflake identifier representing the user who created this emoji; 
+        ///     <c>null</c> if unknown.
+        /// </returns>
+        ulong? CreatorId { get; }
```

Comment:
The API provides a partial user object for this when it can. 

How do we want to handle cases like this? I see a few possibilities in my head.

- Providing both the parent id and parent object props. If the parent object is partial, the prop is null.
- Same as above, but a partial parent object is provided with whatever data was passed from the API. (This may be a little hairy if we also decide to utilize nullable reference types in the lib).
- Only one of the props is provided, most likely Id.